### PR TITLE
Version 4.30.0

### PR DESCRIPTION
**Pull request by qsoftdevelopment**
Version 4.30.0

### CODE REVIEW

File: src/main/java/com/pubnub/api/enums/PNPushType.java [hunk 1]
```diff
@@ -4,6 +4,6 @@
 
     APNS,
     MPNS,
-    GCM
-
+    GCM,
```
Comment (davidnub) [inline src/main/java/com/pubnub/api/enums/PNPushType.java]: Do we need to include `FCM` too?

File: src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    private MPNSPayload mpnsPayload;
+    private APNSPayload apnsPayload;
+
+    public Map<String, Object> build() {
+        Map<String, Object> payload = new HashMap<>();
+
+        if (apnsPayload != null) {
+            Map<String, Object> apnsMap = apnsPayload.toMap();
+            if (!apnsMap.isEmpty()) {
+                payload.put("pn_apns", apnsMap);
```
Comment (davidnub) [inline src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java]: Should we separate APNS vs APNS2?

File: src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            }
+        }
+
+        if (fcmPayload != null) {
+            Map<String, Object> fcmMap = fcmPayload.toMap();
+            if (!fcmMap.isEmpty()) {
+                payload.put("pn_gcm", fcmMap);
```
Comment (davidnub) [inline src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java]: Similarly do we need a separation for FCM/GCM?

**Comment by qsoftdevelopment** [comment]
@davidnub Changes implemented.

File: src/main/java/com/pubnub/api/enums/PNPushType.java (hunk 1 above)
Comment (azurqsd) [reply src/main/java/com/pubnub/api/enums/PNPushType.java]: Yes and no. 

The best solution would be to drop GCM completely and use FCM instead. 

No matter GCM or FCM, in the end it's the same in regards to the SDK functionality, but the docs need to be updated. 

Also a permit is needed to do so.

File: src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java (hunk 2 above)
Comment (azurqsd) [reply src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java]: @davidnub 

Actually it's separated already but the naming isn't that straightforward, which is now improved in another PR #193

File: src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java (hunk 3 above)
Comment (azurqsd) [reply src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java]: @davidnub 

No, they're the same. 

We still have a discrepancy in terms of GCM and FCM being used interchangeably in the SDK. But the logic and functionality remain the same.

**Pull request merged**